command/server.go

```diff
@@ -1641,39 +1641,86 @@ func (c *ServerCommand) Run(args []string) int {
 			logWriter := c.logger.StandardWriter(&hclog.StandardLoggerOptions{})
 			pprof.Lookup("goroutine").WriteTo(logWriter, 2)
 
-			if os.Getenv("VAULT_STACKTRACE_WRITE_TO_FILE") != "" {
-				c.logger.Info("Writing stacktrace to file")
-
-				dir := ""
-				path := os.Getenv("VAULT_STACKTRACE_FILE_PATH")
-				if path != "" {
-					if _, err := os.Stat(path); err != nil {
-						c.logger.Error("Checking stacktrace path failed", "error", err)
-						continue
-					}
-					dir = path
-				} else {
-					dir, err = os.MkdirTemp("", "vault-stacktrace")
-					if err != nil {
-						c.logger.Error("Could not create temporary directory for stacktrace", "error", err)
-						continue
-					}
+			if os.Getenv("VAULT_PPROF_SIGUSR2_DUMP") != "" {
```

Comment:
Ideally, I'd like to see the output format be a debug bundle that can be consumed by debug-helper, and for it to include metrics.  Needn't be part of this PR, just thought I'd mention it so you could consider it.

Whether or not you take that suggestion, another idea to consider: allow an env var (possibly  VAULT_PPROF_SIGUSR2_DUMP itself) to specify an optional duration, and when that's provided, collect cpu profile info for that amount of time, including it as another output file.